### PR TITLE
xcm v1

### PR DESCRIPTION
**Pull request by GopherJ**
close: #376 
close: #618
close: #641 

**TODO** before entering master:

- [x] build & publish parallelfinance/polkadot:v0.9.10 docker image
- [x] test cross chain transfer against the new docker image
- [x] fix loans benchmarks' logic
- [x] add loans benchmarks to runtimes
- [x] test make launch still works
- [x] switch to orml master when xcm v1 PR is merged
- [x] switch to polkadot-release-v0.9.10's final commit
- [x] update changelog
- [x] rename AssetIdentifier to CurrencyId after #639 

**TODO** after entering master:

-  [ ] parallel-js update, don't use AssetId, use CurrencyId instead
-  [ ] parallel-dapp should then use xcm v1 instead of xcm v0

### CODE REVIEW

File: runtime/heiko/src/lib.rs [hunk 1]
```diff
@@ -399,10 +405,11 @@ parameter_types! {
     pub const MetadataDepositPerByte: Balance = deposit(0, 1);
 }
 
-impl pallet_assets::Config for Runtime {
+type AssetsInstance1 = pallet_assets::Instance1;
```
Comment (yz89) [inline runtime/heiko/src/lib.rs:408]: Is there a case that you need to use multiple assets instances?

File: pallets/loans/src/benchmarking.rs [hunk 2]
```diff
@@ -36,28 +37,33 @@ const PENDING_MARKET_MOCK: Market = {
 
 const INITIAL_AMOUNT: u32 = 500_000_000;
 
-fn transfer_initial_balance<T: Config>(caller: T::AccountId)
-where
+fn transfer_initial_balance<
+    T: Config + pallet_assets::Config<AssetId = AssetIdentifier, Balance = Balance>,
```
Comment (yz89) [inline pallets/loans/src/benchmarking.rs:41]: Great work! What I don't understand is that Loans doesn't implement pallet_assets:Config if there is no inheritance. How does this work？

Comment (GopherJ) [reply pallets/loans/src/benchmarking.rs:41]: this is for limiting runtime used for benchmarks, `mock` and all runtimes implemented `pallet_assets::Config` so it worked

File: runtime/vanilla/Cargo.toml [hunk 3]
```diff
@@ -20,71 +20,70 @@ smallvec          = '1.6.1'
...
+orml-oracle-rpc-runtime-api = { git = 'https://github.com/open-web3-stack/open-runtime-module-library.git', branch = 'sw/xcm-v1', default-features = false }
+orml-traits                 = { git = 'https://github.com/open-web3-stack/open-runtime-module-library.git', branch = 'sw/xcm-v1', default-features = false }
+orml-utilities              = { git = 'https://github.com/open-web3-stack/open-runtime-module-library.git', branch = 'sw/xcm-v1', default-features = false }
+orml-vesting                = { git = 'https://github.com/open-web3-stack/open-runtime-module-library.git', branch = 'sw/xcm-v1', default-features = false }
+orml-xcm                    = { git = 'https://github.com/open-web3-stack/open-runtime-module-library.git', branch = 'sw/xcm-v1', default-features = false }
+orml-xcm-support            = { git = 'https://github.com/open-web3-stack/open-runtime-module-library.git', branch = 'sw/xcm-v1', default-features = false }
+orml-xtokens                = { git = 'https://github.com/open-web3-stack/open-runtime-module-library.git', branch = 'sw/xcm-v1', default-features = false }
```
Comment (MrPai) [inline runtime/vanilla/Cargo.toml:86]: need to wait for this to be merged into orml repo.

**Review by MrPai — COMMENTED** [review]
All looks good, let's see when orml merge sw/xcm-v1.

File: runtime/vanilla/Cargo.toml (hunk 3 above)
Comment (GopherJ) [reply runtime/vanilla/Cargo.toml:86]: yes

**Comment by GopherJ** [comment]
I cannot test cross chain yet, the current polkadot apps UI looks like:

image

Blocked by: https://github.com/polkadot-js/apps/issues/6199

**Comment by GopherJ** [comment]
If I use MultiAsset v0 it'll work:

image

**Comment by GopherJ** [comment]
encounterred `InvalidDest` while do upward transfer:

image

**Pull request merged**